components/dropdown/dropdown-content-mixin.js

```diff
@@ -0,0 +1,556 @@
+		return this.shadowRoot.querySelector('.d2l-dropdown-content-container');
+	}
+
+	__getPositionContainer() {
+		return this.shadowRoot.querySelector('.d2l-dropdown-content-position');
+	}
+
+	__getWidthContainer() {
+		return this.shadowRoot.querySelector('.d2l-dropdown-content-width');
+	}
+
+	__getOpener() {
+		const opener = findComposedAncestor(this, (elem) => {
+			if (elem.isDropdownOpener) {
+				return true;
+			}
+		});
+		return opener;
+	}
+
+	__onAutoCloseFocus() {
+
+		/* timeout needed to work around lack of support for relatedTarget */
+		setTimeout(() => {
+			if (!this.opened
+				|| this.noAutoClose
+				|| !document.activeElement
+				|| document.activeElement === this.__previousFocusableAncestor
+				|| document.activeElement === document.body) {
+				return;
+			}
+
+			const activeElement = getComposedActiveElement();
+
+			if (isComposedAncestor(this, activeElement)
+				|| isComposedAncestor(this.__getOpener(), activeElement)) {
+				return;
+			}
+
+			this.opened = false;
+		}, 0);
+	}
+
+	__onAutoCloseClick(e) {
+		if (!this.opened || this.noAutoClose) {
+			return;
+		}
+		const rootTarget = e.composedPath()[0];
+		const content = this.__getContentContainer();
+		if (isComposedAncestor(content, rootTarget)) {
+			return;
+		}
+		const opener = this.__getOpener();
+		if (isComposedAncestor(opener.getOpenerElement(), rootTarget)) {
+			return;
+		}
+
+		this.opened = false;
+	}
+
+	__onClose(e) {
+
+		if (e.target !== this || !document.activeElement) {
+			return;
+		}
+
+		const activeElement = getComposedActiveElement();
+
+		if (!isComposedAncestor(this, activeElement)) {
+			return;
+		}
+
+		const opener = this.__getOpener();
+		opener.getOpenerElement().focus();
+
+	}
+
+	__onResize() {
+		if (!this.opened) {
+			return;
+		}
+		this.__position();
+	}
+
+	async __openedChanged(newValue) {
+
+		this.__previousFocusableAncestor =
+			newValue === true
+				? getPreviousFocusableAncestor(this, false, false)
+				: null;
+
+		this.__isRTL = (getComputedStyle(this).direction === 'rtl');
+
+		const doOpen = () => {
+
+			const content = this.__getContentContainer();
+
+			if (!this.noAutoFit) {
+				content.scrollTop = 0;
+			}
+
+			this.__position(undefined, undefined);
+
+			if (!this.noAutoFocus && this.__applyFocus) {
+				const focusable = getFirstFocusableDescendant(this);
+				if (focusable) {
+					// bumping this to the next frame is required to prevent IE/Edge from crazily invoking click on the focused element
+					requestAnimationFrame(() => {
+						focusable.focus();
+					});
+				} else {
+					content.setAttribute('tabindex', '-1');
+					content.focus();
+				}
+			}
+
+			this.dispatchEvent(new CustomEvent('d2l-dropdown-open', { bubbles: true, composed: true }));
+
+			this.__dismissibleId = setDismissible(() => {
+				this.close();
+			});
+		};
+
+		if (newValue) {
+			if (!this.renderContent) {
+				await this.forceRender();
+			} else {
+				await this.updateComplete;
+			}
+
+			doOpen();
+

```

Comment:
[The mutation observer](https://github.com/BrightspaceUI/dropdown/blob/master/d2l-dropdown-content-behavior.js#L554) will be re-added in the pull request that converts the `d2l-dropdown-tabs` and `d2l-dropdown-menu` components.